unittests/test_meta.py

```diff
@@ -201,3 +203,22 @@ def hook_b(self):
     assert not Bar.hook_in_stage('hook_b', 'pre_compile')
     assert Bar.hook_in_stage('hook_c', 'post_run_wait')
     assert Bar.hook_in_stage('hook_a', 'pre_sanity')
+
+
+def test_final(MyMeta):
+    class Base(MyMeta):
+        @final
+        def foo(self):
+            pass
+
+    with pytest.raises(ReframeSyntaxError):
+        class Derived(Base):
+            def foo(self):
+                '''Override attempt.'''
+
+    class AllowFinalOverride(Base):
+        '''Use flag to bypass the final override check.'''
+        _rfm_override_final = True
```

Comment:
Why not like this, which is the documented way?

```suggestion
    class AllowFinalOverride(Base, special=True):
        '''Use flag to bypass the final override check.'''
```

Reply:
This test just checks that the metaclass machinery to override `final` methods works. The `special=True` belongs to the API defined in the `RegressionTest` class and should not be tested for here. That check is already in `unittests/test_loader.py`.